compiler/rustc_codegen_llvm/src/debuginfo/metadata.rs

```diff
@@ -1525,7 +1526,7 @@ fn prepare_union_metadata<'ll, 'tcx>(
 /// on MSVC we have to use the fallback mode, because LLVM doesn't
 /// lower variant parts to PDB.
 fn use_enum_fallback(cx: &CodegenCx<'_, '_>) -> bool {
-    cx.sess().target.is_like_msvc
+    cpp_like_debuginfo(cx.tcx)
```

Comment:
I should probably just inline this function and comment.

Reply:
That seems like a good idea.